File: blackJack.py
from cards import cards
import random

print("Welcome to BlackJack!")

playAgain = 'y'
while playAgain == 'y':
  print("\n")
  myCards = []
  dealerCard = random.choice(cards)
  print("Dealers Card:", dealerCard)
  card1 = random.choice(cards)
  # card1 = 11
  card2 = random.choice(cards)
  myCards.append(card1)
  myCards.append(card2)
  total = card1 + card2
  print("Here are your cards: ")
  print(*myCards, sep=" ")
  print("Total:", total)
  hit = input("Hit? y/n ")
  while total < 21 and hit == 'y':
    total = 0
    print("Here are your cards: ")
    card1 = random.choice(cards)
    myCards.append(card1)
    for card in myCards:
      total += card
    if total > 21:
      if 11 in myCards:
        total = 0
        for i in range(len(myCards)):
          if myCards[i] == 11:
            myCards[i] = 1
        for card in myCards:
          total += card
      else:
        hit = 'n'
    
    print(*myCards, sep=" ")
    print("Total", total)
    print("\n")
    if total == 21:
      hit = 'n'
    else:
      hit = input("Hit? y/n ")

  print("Your final total is:", total)
  dealerTotal = dealerCard
  broke = False
  while not broke:
    if dealerTotal < 17:
      newCard = random.choice(cards)
      dealerTotal += newCard
    if dealerTotal > 16:
      broke = True
  print("Dealer Total:", dealerTotal)
  
  if total > 21:
    print("You broke! You lost!")
  elif dealerTotal > 21:
    print("Dealer broke! YOU WIN!!")
  elif total > dealerTotal:
    print("YOU WIN!")
  else: 
    print("You lost!")


  playAgain = input("Play again? y/n ")

File: blackJack.java
import java.util.Random;
import java.util.Scanner;
import java.util.*;

class Main {

  public static void main(String[] args) {

    Vector deck = new Vector();

    for (int i = 1; i <= 11; i++) {
      for (int j = 0; j < 4; j++) {
        deck.add(i);
        if (i == 10) {
          deck.add(i);
          deck.add(i);
        }
      }
    }

    System.out.println("Welcome to BLACKJACK!");
    Scanner scanner = new Scanner(System.in);
    System.out.println("Are you ready to play? y/n ");
    String userInput = scanner.nextLine();

    Random rand = new Random();
    int card1_index = 0;
    int card2_index = 0;
    int card1 = 0;
    int card2 = 0;
    int dealerCard = 0;
    
    while (userInput.equals("y")) {
      System.out.println("ENTER WHILE LOOP");
      Vector myCards = new Vector();
      int upperBound = deck.size();
      card1_index = rand.nextInt(upperBound);

      // card1 = deck.get(card1_index);
      deck.remove(card1_index);
      upperBound--;
      card2_index = rand.nextInt(upperBound);
      // card2 = deck.get(card2_index);
      deck.remove(card2_index);
      upperBound--;
      System.out.println("Dealer ")



      userInput = scanner.nextLine();





    }


    // 17 or higher stay
    // 16 or lower must hit
    // Scanner scanner = new Scanner(System.in);
    // String playAgain = "y";
    // while (playAgain == 'y') {
    //   // receive two cards
    //   Random rand = new Random();
    //   int upperBound = cards.size();
      



      






  }
}